src/library_wasmfs_opfs.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * @license
+ * Copyright 2022 The Emscripten Authors
+ * SPDX-License-Identifier: MIT
+ */
+
+mergeInto(LibraryManager.library, {
+  $wasmfsOPFSDirectories: {
+    allocated: [],
+    free: [],
+    get: function(i) {
+      return this.allocated[i];
+    }
+  },
+
+  $wasmfsOPFSFiles: {
+    allocated: [],
+    free: [],
+    get: function(i) {
+      return this.allocated[i];
+    }
+  },
+
+  $wasmfsOPFSAccesses: {
+    allocated: [],
+    free: [],
+    get: function(i) {
+      return this.allocated[i];
+    }
+  },
```

Comment:
These three could all be generated from a single function that creates such instances. A TODO to refactor though might be enough for now. It's possible we have other such stuff that could be refactored with them too.